Searches/BinarySearch.java

```diff
@@ -45,7 +45,7 @@
     if (right < left) return -1; // this means that the key not found
 
     // find median
-    int median = left + ((right - left) >>> 1);
+    int median = left + ((right - left) / 2);
```

Comment:
```suggestion
    int median = (left + right) >>> 1;
```

Reply:
What about `int median = (left + right) / 2;`?
After all, the whole issue here is the `>>> 1` part...

Reply:
When `left = Integer.MAX_VALUE - 1`, `right = Integer.MAX_VALUE`,  `int median = (left + right) / 2` will not work exactly.